### PR TITLE
Add custom cursor tutorial

### PR DESCRIPTION
**Pull request by oliverdunk**
Adds the custom cursor extension from an upcoming YouTube video.

### CODE REVIEW

File: functional-samples/tutorial.custom-cursor/README.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# Custom Cursor extension
+
+Adds a custom cursor to pages on https://developer.chrome.com. This sample is part of a tutorial on YouTube.
```
Comment (patrickkettner) [inline functional-samples/tutorial.custom-cursor/README.md:3]: were you planning on adding a link to the tutorial once its live?

File: functional-samples/tutorial.custom-cursor/counter.js [hunk 2]
```diff
@@ -0,0 +1,3 @@
+document.addEventListener('click', async () => {
```
Comment (patrickkettner) [inline functional-samples/tutorial.custom-cursor/counter.js:1]: why is it async if you aren't using the return value?

File: functional-samples/tutorial.custom-cursor/service-worker.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+chrome.runtime.onMessage.addListener(async (msg) => {
+  if (msg === 'page-click') {
+    const data = await chrome.storage.local.get({ counter: 0 });
+    let counter = data['counter'];
+    counter++;
+    chrome.storage.local.set({ counter });
+    await chrome.action.setBadgeText({ text: counter.toString() });
```
Comment (patrickkettner) [inline functional-samples/tutorial.custom-cursor/service-worker.js:7]: is the toString call needed? Don't care either way, id just be surprised if it was

**Review by patrickkettner — APPROVED** [review]
couple of questions, but LGTM! Super cute

File: functional-samples/tutorial.custom-cursor/service-worker.js (hunk 3 above)
Comment (oliverdunk) [reply functional-samples/tutorial.custom-cursor/service-worker.js:7]: It is actually!

Screenshot 2024-07-30 at 14 59 37

File: functional-samples/tutorial.custom-cursor/README.md (hunk 1 above)
Comment (oliverdunk) [reply functional-samples/tutorial.custom-cursor/README.md:3]: Yes, definitely. I considered blocking this PR on it but I think I might just merge now and open a follow-up PR when the video is live.

File: functional-samples/tutorial.custom-cursor/counter.js (hunk 2 above)
Comment (oliverdunk) [reply functional-samples/tutorial.custom-cursor/counter.js]: Good spot, thanks! That wasn't needed. Went ahead and removed it.

File: functional-samples/tutorial.custom-cursor/service-worker.js (hunk 3 above)
Comment (patrickkettner) [reply functional-samples/tutorial.custom-cursor/service-worker.js:7]: my loosey-goosey-everything-is-an-object-anyway-javascript background is deeply offended

File: functional-samples/tutorial.custom-cursor/README.md (hunk 1 above)
Comment (patrickkettner) [reply functional-samples/tutorial.custom-cursor/README.md:3]: sgtm

**Pull request merged**